### PR TITLE
[INFINITY-932] Update dcos-commons to reflect upstreamed test code. Avoid GCing all data.

### PR DESCRIPTION
**Pull request by nickbp**
Follows up on https://github.com/dcos/shakedown/pull/116 and https://github.com/dcos/shakedown/pull/119

### CODE REVIEW

**Comment by nickbp** [comment]
retest this please (more flakes from apache.org)

**Comment by nickbp** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please (apparently jenkins died? so close!!!!!!!!!)

**Comment by mbdcos** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please (hung again!)

**Comment by nickbp** [comment]
if this is still hanging in the morning it's getting merged. nothing's failed for the last 24h of retesting

**Comment by mbdcos** [comment]
Can we ask Jenkins team to figure out why it is hanging? May be running out of memory, may be a timer, may be some internal task is failing. It does not look like a temporary problem.

File: frameworks/helloworld/tests/test_discovery.py [hunk 1]
```diff
@@ -31,5 +34,5 @@ def test_task_dns_prefix_points_to_all_tasks():
     # Assert that DiscoveryInfo is correctly set on tasks.
     assert(all(p["info"]["discovery"]["name"] == "hello-0" for p in pod_info))
     # Assert that the hello-0.hello-world.mesos DNS entry points to the right IP.
-    spin.time_wait_noisy(lambda: (
-        plan.get_deployment_plan(PACKAGE_NAME).json()['status'] == 'COMPLETE'))
+    shakedown.wait_for(lambda: (
+        plan.get_sidecar_plan(PACKAGE_NAME).json()['status'] == 'COMPLETE'))
```
Comment (loren) [inline frameworks/helloworld/tests/test_discovery.py]: I think your problem is here. You are waiting for a plan named `sidecar` but `discovery.yml` only has `deploy` specified and `recovery` implicitly.

Comment (mbdcos) [inline frameworks/helloworld/tests/test_discovery.py]: Wait_for will timeout and it will generate an error, isn't it?

Comment (nickbp) [inline frameworks/helloworld/tests/test_discovery.py]: Not sure why I changed this, bad merge? Worth a shot in any case.

**Comment by nickbp** [comment]
retest this please

**Comment by nickbp** [comment]
Interest in keeping this constantly rebased is waning...

**Pull request closed without merging**